rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -117,27 +134,40 @@ def _parse_step(self, step: Dict[Text, Any]) -> None:
             )
 
     def _parse_user_utterance(self, step: Dict[Text, Any]) -> None:
+        utterance = self._parse_raw_user_utterance(step)
+        if utterance:
+            self._validate_that_utterance_is_in_domain(utterance)
+            self.current_step_builder.add_user_messages([utterance])
 
-        if self.use_e2e:
-            # TODO
-            pass
-        else:
-            utterance = self._parse_raw_user_utterance(step)
-            if utterance:
-                self.current_step_builder.add_user_messages([utterance])
+    def _validate_that_utterance_is_in_domain(self, utterance: UserUttered) -> None:
+        intent_name = utterance.intent.get("name")
+
+        if not self.domain:
+            logger.debug(
+                "Skipped validating if intent is in domain as domain " "is `None`."
+            )
+            return
+
+        if intent_name not in self.domain.intents:
+            common_utils.raise_warning(
+                f"Issue found in '{self.source_name}': \n"
+                f"Found intent '{intent_name}' in stories which is not part of the "
+                f"domain.",
```

Comment:
Should we also mention that this utterance will be skipped then?

Reply:
I don't skip it. I just issue a warning for now.

Reply:
AFAIK the same check will happen later and it will be skipped during the training anyway. But I'm fine with it.

Reply:
I tried to find it, but I think only `rasa data validate` will check, right?

Reply:
Markdown also just warns and continues